src/Docker/IntegrationV2Builder.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Docker;
+
+use Illuminate\Contracts\Config\Repository;
+use Magento\MagentoCloud\Docker\Service\ServiceFactory;
+use Magento\MagentoCloud\Filesystem\FileList;
+
+/**
+ * Docker integration test builder.
+ *
+ * @codeCoverageIgnore
+ */
+class IntegrationV2Builder implements BuilderInterface
+{
+    /**
+     * @var FileList
+     */
+    private $fileList;
+
+    /**
+     * @var ServiceFactory
+     */
+    private $serviceFactory;
+
+    /**
+     * @param FileList $fileList
+     * @param ServiceFactory $serviceFactory
+     */
+    public function __construct(FileList $fileList, ServiceFactory $serviceFactory)
+    {
+        $this->fileList = $fileList;
+        $this->serviceFactory = $serviceFactory;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
```

Comment:
you may move configurations for fpm, db, web services to separate private methods and collect them here. It'll make code more easy to read/change and you will avoid @SuppressWarning

Reply:
This is not customer-facing code

Reply:
yes, we do code more readable for your convenience.. But I don't insist